manifests/init.pp

```diff
@@ -91,105 +91,107 @@
-  Optional[Boolean]                 $redis_slave_role                = undef,
-  Optional[Boolean]                 $redis_sentinel_role             = undef,
-  Optional[Hash]                    $registry                        = undef,
-  Optional[String]                  $registry_external_url           = undef,
-  Optional[Hash]                    $registry_nginx                  = undef,
-  Boolean                           $registry_nginx_eq_nginx         = false,
-  Optional[Hash]                    $repmgr                          = undef,
-  Optional[Array]                   $roles                           = undef,
-  Optional[Hash]                    $sentinel                        = undef,
-  Boolean                           $skip_post_deployment_migrations = false,
-  Optional[Hash]                    $shell                           = undef,
-  Optional[Hash]                    $sidekiq                         = undef,
-  Optional[Hash]                    $sidekiq_cluster                 = undef,
-  Enum['present', 'absent']         $skip_auto_reconfigure           = 'absent',
-  Optional                          $skip_auto_migrations            = undef,
-  Optional[Stdlib::Absolutepath]    $source_config_file              = undef,
-  Boolean                           $store_git_keys_in_db            = false,
-  Optional[Hash]                    $unicorn                         = undef,
-  Optional[Hash]                    $puma                            = undef,
-  Optional[Hash]                    $gitlab_workhorse                = undef,
-  Optional[Hash]                    $user                            = undef,
-  Optional[Hash]                    $web_server                      = undef,
-  Boolean                           $backup_cron_enable              = false,
-  Integer[0,59]                     $backup_cron_minute              = 0,
-  Integer[0,23]                     $backup_cron_hour                = 2,
-  Array                             $backup_cron_skips               = [],
-  Hash                              $custom_hooks                    = {},
-  Hash                              $global_hooks                    = {},
+  String                              $rake_exec                       = '/usr/bin/gitlab-rake',
+  Optional[Hash]                      $alertmanager                    = undef,
+  Optional[Hash]                      $ci_redis                        = undef,
+  Optional[Hash]                      $ci_unicorn                      = undef,
+  Boolean                             $config_manage                   = true,
+  Stdlib::Absolutepath                $config_file                     = '/etc/gitlab/gitlab.rb',
+  Optional[Hash]                      $consul                          = undef,
+  Optional[String]                    $custom_hooks_dir                = '/opt/gitlab/embedded/service/gitlab-shell/hooks',
```

Comment:
since those two variables now have a default value, they cannot be Optional (because you cannot pass undef to it).
when when we already update it we can update the datatype as well
```suggestion
  Stdlib::Absolutepath                $custom_hooks_dir                = '/opt/gitlab/embedded/service/gitlab-shell/hooks',
```